common_utils/artifacts_utils.py

```diff
@@ -0,0 +1,44 @@
+import os
+
+def get_latest_model_artifact(path, run_type):
+    """
+    Retrieve the latest model artifact for a given run type based on the modification time.
+
+    Args:
+        path (str): The model specifc directory path where artifacts are stored. 
+        Where PATH_ARTIFACTS = setup_artifacts_paths(PATH) executed in the model specifc main.py script.
+        and PATH = Path(__file__)
+        
+        run_type (str): The type of run (e.g., calibration, testing, forecasting).
+
+    Returns:
+        str: The path to the latest model artifact given the run type.
+
+    Raises:
+        FileNotFoundError: If no model artifacts are found for the given run type.
+    """
+
+    # List all model files for the given specific run_type with the expected filename pattern
+    model_files = [f for f in os.listdir(path) if f.startswith(f"{run_type}_model_") and f.endswith('.pt')]
```

Comment:
Okay I see it below

Reply:
I have created a new function to take into account the most common file extensions - more can be added as needed.
Right now it accommodates: '.pt', '.pth', '.h5', '.hdf5', '.pkl', '.json', '.bst', '.txt', '.bin', '.cbm', '.onnx'
I have not tested the pipeline with anything but .pt, so that could be issues with one or more of the extensions above elsewhere in the pipeline - time will tell